### PR TITLE
(SLV-275) SPIKE - Investigate using Ruby Task Helper for RAS tasks

### PR DESCRIPTION
**Pull request by billclaytor**
* Note: This PR is not compatible with the update to Bolt 1.0 which uses bolt as a package and removes the ability to install modules from the forge. The task helpers are included in Bolt with version 1.2 so once we upgrade installing from the forge will no longer be required.

This PR includes the `pe_versions` example task using the Bolt Ruby Task Helper:
https://github.com/puppetlabs/puppetlabs-ruby_task_helper

pe_versions - a task that outputs the PE versions compatible with RAS
```
user:~/RubymineProjects/ref_arch_setup> bolt task run ref_arch_setup::pe_versions -n localhost --modulepath ./modules
Started on localhost...
Finished on localhost:
  Checking Puppet Enterprise Version History: https://puppet.com/misc/version-history
  RAS supports the following PE versions:
  2019.0.1
  2018.1.5
  2019.0.0
  2018.1.4
  2018.1.3
  2018.1.2
  2018.1.1
  2018.1.0
  
  {"versions":["2019.0.1","2018.1.5","2019.0.0","2018.1.4","2018.1.3","2018.1.2","2018.1.1","2018.1.0"]}
  {
  }
Successful on 1 node: localhost
Ran on 1 node in 1.09 seconds
```

### CODE REVIEW

File: modules/ref_arch_setup/tasks/pe_versions.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+require_relative "../../ruby_task_helper/lib/task_helper.rb"
+
+# PE Versions Task using Ruby Task Helper
+class VersionsTask < TaskHelper
+  PE_VERSIONS_URL = "https://puppet.com/misc/version-history".freeze
+
+  # the minimum prod version supported by RAS
+  MIN_PROD_VERSION = "2018.1.0".freeze
+
+  def task(**_kwargs)
```
Comment (RandellP) [inline modules/ref_arch_setup/tasks/pe_versions.rb]: what is up with the **_kwargs

File: modules/ref_arch_setup/tasks/pe_versions.rb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    return is_valid_response
+  end
+  # rubocop:enable Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
+  # rubocop:enable Metrics/CyclomaticComplexity
+  # rubocop:enable Metrics/PerceivedComplexity
+end
+
+VersionsTask.run if $PROGRAM_NAME == __FILE__
```
Comment (RandellP) [inline modules/ref_arch_setup/tasks/pe_versions.rb]: I don't see a run method defined...

File: modules/ref_arch_setup/tasks/slv_275.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (RandellP) [inline modules/ref_arch_setup/tasks/slv_275.json]: I assume you didn't mean to include this file

File: modules/ref_arch_setup/tasks/slv_275.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env ruby
```
Comment (RandellP) [inline modules/ref_arch_setup/tasks/slv_275.rb]: I assume you didn't mean to include this file

File: Boltdir/Puppetfile [hunk 5]
```diff
@@ -1,2 +1,3 @@
 # Modules from the Puppet Forge
 mod "puppetlabs-facts", "0.2.0"
+mod "puppetlabs-ruby_task_helper", "0.1.0"
```
[the remark this answers is not in the document]
Comment (billclaytor) [reply Boltdir/Puppetfile]: Updated

File: modules/ref_arch_setup/tasks/pe_versions.rb (hunk 1 above)
Comment (billclaytor) [reply modules/ref_arch_setup/tasks/pe_versions.rb]: This is directly from the example:
https://github.com/puppetlabs/puppetlabs-ruby_task_helper

I was wondering the same thing when I saw it and found this explanation:
https://flushentitypacket.github.io/ruby/2015/03/31/ruby-keyword-arguments-the-double-splat-and-starsnake.html

"Keyword arguments give you a convenient shorthand for using hashes as an argument to a method. ... There is a new operator, the double splat, that is sort of the splat for keyword arguments."

File: modules/ref_arch_setup/tasks/pe_versions.rb (hunk 2 above)
Comment (billclaytor) [reply modules/ref_arch_setup/tasks/pe_versions.rb]: VersionsTask inherits from TaskHelper:
https://github.com/puppetlabs/puppetlabs-ruby_task_helper/blob/master/lib/task_helper.rb

File: modules/ref_arch_setup/tasks/slv_275.json (hunk 3 above)
Comment (billclaytor) [reply modules/ref_arch_setup/tasks/slv_275.json]: I just included it as a basic example similar to the example in the doc. It is not intended to be merged.

Comment (billclaytor) [reply modules/ref_arch_setup/tasks/slv_275.json]: Removed

File: modules/ref_arch_setup/tasks/slv_275.rb (hunk 4 above)
Comment (billclaytor) [reply modules/ref_arch_setup/tasks/slv_275.rb]: Removed

**Pull request merged**